Show user courses for sensei_user_courses shortcode in different languages

merkushin opened the pull request:
Resolves https://github.com/Automattic/sensei/pull/7492#pullrequestreview-1978611313

## Proposed Changes
* Show user courses for sensei_user_courses shortcode in different languages.

## Known Issues

* This PR doesn't affect Course List block. Can't find a solution for it at the moment.
* Now we actually enrol a user into each translated course. The drawback of this solution is that the user is not automatically enrolled in a new translation. Could be addressed later as part of translation creation. (If it is okay, I'll create a separate issue.)

## Testing Instructions

1. On the My Courses page replace Course List block with the `[sensei_user_courses]` shortcode.
2. Translate My Courses page.
3. Follow our instructions to create one or more courses and their translations.
4. Go to the frontend of the original language and take one or more courses.
5. Switch to another language on the frontend.
6. Check that on My Courses page you see all the courses you took and their current (shared) status. (Notice: you won't see a course if it doesn't have a translation.)

## New/Updated Hooks

* `sensei_learner_enrolled_courses_query_by_progress_status_course_ids` - Filters the course IDs for a learner's enrolled courses query by progress status.
* `sensei_settings_my_course_page_id` - Filters the My Courses page ID.

## Removed Hooks (Not released yet)
- `sensei_course_manual_enrolment_enroll_learner_course_id`
- `sensei_course_manual_enrolment_withdraw_learner_course_id`

## Pre-Merge Checklist
<!-- Complete applicable items on this checklist **before** merging. Items that are not applicable can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed. -->
- [x] PR title and description contain sufficient detail and accurately describe the changes
- [ ] Acceptance criteria is met
- [x] Decisions are publicly documented
- [x] Adheres to coding standards ([PHP](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/), [JavaScript](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/), [CSS](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/), [HTML](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/html/))
- [ ] All strings are translatable (without concatenation, handles plurals)
- [x] Follows our naming conventions (P6rkRX-4oA-p2)
- [x] Hooks (p6rkRX-1uS-p2) and functions are documented
- [ ] New UIs are responsive and use a [mobile-first approach](https://zellwk.com/blog/how-to-write-mobile-first-css/)
- [ ] New UIs match the designs
- [ ] Different user privileges (admin, teacher, subscriber) are tested as appropriate
- [ ] Legacy courses (course without blocks) are tested
- [ ] Code is tested on the minimum supported PHP and WordPress versions
- [ ] User interface changes have been tested on the latest versions of Chrome, Firefox and Safari
- [ ] "Needs Documentation" label is added if this change requires updates to documentation
- [ ] Known issues are created as new GitHub issues

merkushin committed:
Add config for course-outline-lesson block

merkushin committed:
Try to configure blocks

merkushin committed:
Fix course outline for a course translation

merkushin committed:
Create translations for lessons and update relations for new lessons

merkushin committed:
Restore upadate lesson properties on lesson translation created

merkushin committed:
Add changelog entry

merkushin committed:
Fix linter issues

merkushin committed:
Fix psalm issues

merkushin committed:
Fix docblock return type

merkushin committed:
Ignore issues in legacy code

merkushin committed:
Add tests for wpml hooks

merkushin committed:
Test lesson course was updated for translated lessons

merkushin committed:
Merge branch 'trunk' into fix/wpml-save-dependent-posts

merkushin committed:
Fix linter issues in tests

merkushin committed:
Merge branch 'trunk' into fix/wpml-save-dependent-posts

merkushin committed:
Handle translations for quizzes and questions, refactor code structure

merkushin committed:
Add hooks, fix some legacy linter issues

merkushin committed:
Update WPML instance, fix legacy linter issues

merkushin committed:
Update Psal baseline

merkushin committed:
Add config for question block

merkushin committed:
Add/fix some tests for WPML

merkushin committed:
Add changelog entry

merkushin committed:
Update tests

merkushin committed:
Update Psal baseline

merkushin committed:
Update Psal baseline

merkushin committed:
Update quiz id for questions

merkushin committed:
WPML: Add support for reusable student progress

merkushin committed:
Merge branch 'trunk' into fix/wpml-quiz

merkushin committed:
Update lesson relations for existing lessons as well

merkushin committed:
Add filters for identifiers in quiz submission repositories

merkushin committed:
Merge branch 'trunk' into fix/wpml-quiz

merkushin committed:
Rename the hook on quiz create

merkushin committed:
Save lesson order for translated lessons

merkushin committed:
Fix usage of do_action_deprecated

merkushin committed:
Merge branch 'fix/wpml-quiz' into fix/wpml-progress

merkushin committed:
Merge branch 'trunk' into fix/wpml-progress

merkushin committed:
Fix linter issue

merkushin committed:
Filter course id when enroll course

merkushin committed:
Make it possible to start translated course and share course and lesson progress

merkushin committed:
Fix comment fetching issues

merkushin committed:
Update tests

merkushin committed:
Add changelog entry

merkushin committed:
Fix linter issues

merkushin committed:
Fix hook docs

merkushin committed:
Fix psalm issues

merkushin committed:
Merge branch 'trunk' into fix/wpml-progress

merkushin committed:
Support question categories

merkushin committed:
Support question categories

merkushin committed:
Add changelog entry and fix linter issues

merkushin committed:
Add changelog entry and fix linter issues

merkushin committed:
Update baseline

merkushin committed:
Update baseline

merkushin committed:
Add tests

merkushin committed:
Add tests

merkushin committed:
Fix linter issues

merkushin committed:
Fix linter issues

merkushin committed:
Extract WPML API and create duplicates when they are needed

merkushin committed:
Extract WPML API and create duplicates when they are needed

merkushin committed:
Fix linter issues

merkushin committed:
Fix linter issues

merkushin committed:
Fix psalm issues

merkushin committed:
Fix psalm issues

merkushin committed:
Create duplicates for questions in category

merkushin committed:
Fix psalm issues

merkushin committed:
Add settings for WPML, return untranslated slugs when setting enabled

merkushin committed:
Fix linter issues

merkushin committed:
Test translation of questions inside a category

merkushin committed:
Clean up code and improve wording

merkushin committed:
Add changelog entry

merkushin committed:
Omit defaults in hooks registration

merkushin committed:
Fix hook name

Co-authored-by: Imran Hossain <imranh920@gmail.com>

merkushin committed:
Explicitly cast lesson id and quiz id to int for consistency

merkushin committed:
Merge branch 'fix/wpml-progress' into fix/wpml-questions-from-category

merkushin committed:
Merge branch 'trunk' into fix/wpml-progress

merkushin committed:
Merge branch 'fix/wpml-progress' into fix/wpml-question-categories

merkushin committed:
Merge branch 'fix/wpml-question-categories' into fix/existing-translation-directions

merkushin committed:
Merge branch 'fix/existing-translation-directions' into fix/wpml-questions-from-category

merkushin committed:
Merge branch 'fix/wpml-questions-from-category' into fix/wpml-translate-slugs

merkushin committed:
Try to create a translation for the category if it doesn't exist yet

merkushin committed:
Merge branch 'fix/wpml-questions-from-category' into fix/wpml-translate-slugs

merkushin committed:
Provide translation ID

merkushin committed:
Merge branch 'fix/wpml-questions-from-category' into fix/wpml-translate-slugs

merkushin committed:
Show active courses for sensei_user_courses shortcode in different languages.

merkushin committed:
Fix hook doc

merkushin committed:
Add changelog entry

merkushin committed:
Fix linter issues

merkushin committed:
Fix linter issues

merkushin committed:
Merge branch 'trunk' into fix/wpml-progress

merkushin committed:
Merge branch 'fix/wpml-progress' into fix/wpml-question-categories

merkushin committed:
Merge branch 'fix/wpml-question-categories' into fix/existing-translation-directions

merkushin committed:
Merge branch 'fix/existing-translation-directions' into fix/wpml-questions-from-category

merkushin committed:
Merge branch 'fix/wpml-questions-from-category' into fix/wpml-translate-slugs

merkushin committed:
Merge branch 'fix/wpml-translate-slugs' into fix/wpml-my-courses

Imran92 reviewed (commented):
Now I am seeing translated Courses in My Courses page when I switch language! 🙌 But After I completed a translated Course, I didn't see it in the completed list

https://github.com/Automattic/sensei/assets/6820724/2cb944c3-e904-4f97-be4b-6ca993d33536

merkushin committed:
Filter course ids when they are fetched for a user by status

merkushin commented:
Thanks @Imran92. 

Fixed it here: https://github.com/Automattic/sensei/pull/7583/commits/c0a5fc9270c3184be0ff8a50cddc3fa0e80508e7

While preparing the PR, moved the filter to a later point and checked only Active tab. Sorry.

Imran92 reviewed (dismissed):
LGTM! 👍

merkushin committed:
Merge branch 'trunk' into fix/wpml-my-courses

merkushin committed:
Fix variable name after merge
